modules/aws/src/main/scala/com.snowplowanalytics.snowplow.lakes/AwsApp.scala

```diff
@@ -103,4 +105,15 @@ object AwsApp extends LoaderApp[KinesisSourceConfig, KinesisSinkConfig](BuildInf
       case None =>
         t.getMessage
     }
+
+  /**
+   * Extracts s3:ActionName from exception message if possible
+   *
+   * Example exception message is as below: User: arn:aws:sts::...loader is not authorized to
+   * perform: s3:PutObject on resource: "arn:aws:s3:::bucket-name/....json" because no
+   * identity-based policy allows the s3:PutObject action (Service: S3, Status Code: 403, Request
+   * ID: req-id, Extended Request ID: ext-req-id)
+   */
+  private def extractS3Action(s3Exception: S3Exception): Option[String] =
+    s3Exception.getMessage.split(" ").filter(_.startsWith("s3:")).distinct.headOption
```

Comment:
Matching only on `s3:` doesn't sound very strict. Also, it doesn't feel good to have `.distinct.headOption` and have a `fold` on the result. Here is an alternative with a regex:
```suggestion
  private def extractMissingPermission(e)(s3Exception: S3Exception): String = {
    val pattern = """.*is not authorized to perform: s3:(\w+).*""".r
    s3Exception.getMessage match {
      case pattern(action) =>
        s"Missing s3:$action permission to setup the destination table in S3 bucket"
      case _ =>
        "Missing permission to setup the destination table in S3 bucket"
  }
```
(not tested)